### PR TITLE
Add @mlflow/vercel for better Vercel AI SDK tracing in Databricks UC

### PR DESCRIPTION
**Pull request by dbrx-euirim**
### What changes are proposed in this pull request?

This PR introduces `@mlflow/vercel`, a lightweight npm package that lets Vercel AI SDK
  users send traces to MLflow (OSS and Databricks) with full feature support
  (structured inputs/outputs, span types, token usage, chat UI rendering, etc.).

#### Why this is needed

The Vercel AI SDK emits OpenTelemetry spans with its own `ai.*` attributes. MLflow's tracing UI and query engine expect `mlflow.*` attributes to power features like the chat message viewer, token cost tracking, and span categorization. The OSS MLflow backend server handles this translation on the read path, but the Databricks write path (the OTel collector endpoint) does not; spans arrive as-is. Without translation, Vercel AI SDK traces land in MLflow missing key metadata.

#### How `@mlflow/vercel` works

 The package exports `MLflowSpanProcessor`, an OpenTelemetry `SpanProcessor` that
  translates Vercel AI SDK span attributes into their MLflow equivalents in onEnd()
  before delegating to an internal `BatchSpanProcessor` for export. It handles span
  types, model info, structured inputs/outputs, token usage, and message format. Spans
  that don't come from the AI SDK (like plain HTTP spans) pass through untouched.

#### Usage example

For an existing Vercel AI SDK app that already sends OTel traces, the only change is wrapping the exporter:

```typescript
  import { MLflowSpanProcessor } from "@mlflow/vercel";
  import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-proto";
  import { NodeTracerProvider } from "@opentelemetry/sdk-trace-node";
  import { generateText } from "ai";
  import { openai } from "@ai-sdk/openai";

  const provider = new NodeTracerProvider({
    spanProcessors: [
      new MLflowSpanProcessor(
        new OTLPTraceExporter({
          url: `${process.env.DATABRICKS_HOST}/api/2.0/otel/v1/traces`,
          headers: {
            Authorization: `Bearer ${process.env.DATABRICKS_TOKEN}`,
            "X-Databricks-UC-Table-Name": "catalog.schema.spans",
          },
        }),
      ),
    ],
  });
  provider.register();

  const result = await generateText({
    model: openai("gpt-4o"),
    prompt: "Summarize the latest ML research trends.",
    experimental_telemetry: { isEnabled: true },
  });
```

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

The package includes comprehensive unit tests for both the exporter wrapper and the attribute translation logic, covering LLM spans, tool calls, embeddings, token usage extraction, and edge cases (missing attributes, pre-existing `mlflow.*` attributes, non-AI-SDK spans). Manual testing was done by running an agentic Vercel AI SDK app against a live Databricks workspace and verifying traces rendered correctly in the MLflow UI.

### Does this PR require documentation update?

- [ ] No.
- [x] Yes. I've updated:
  - [x] Examples
  - [ ] API references
  - [ ] Instructions

The package includes a README with setup instructions and API documentation.

### Does this PR require updating the [MLflow Skills](https://github.com/mlflow/skills) repository?

- [x] No.
- [ ] Yes. Please link the corresponding PR or explain how you plan to update it.

#### Is this a user-facing change?

- [ ] No.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Added `@mlflow/vercel`, a new npm package that enables Vercel AI SDK users to send fully-featured traces to MLflow on Databricks. Wrap your existing OTel exporter with `MLflowSpanExporter` to automatically translate AI SDK span attributes into the format MLflow expects.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Is this a critical bugfix or security fix that should go into the next patch release?

- [ ] This PR is critical and needs to be in the next patch release
- [x] This PR can wait for the next minor release

### CODE REVIEW

File: libs/typescript/integrations/vercel/package.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "@mlflow/vercel",
+  "version": "0.2.0",
```
Comment (dbrx-euirim) [inline libs/typescript/integrations/vercel/package.json]: FYI @B-Step62 I started at 0.2.0 to match other integrations. I'm not sure if I should start at 0.1.0

Comment (B-Step62) [reply libs/typescript/integrations/vercel/package.json]: You're right, let's start with 0.2.0 to align. Will update to `0.2.0-rc.0` to release rc.

**Review by B-Step62 — APPROVED** [review]
LGTM, I updated the code to use span processor rather than exporter, because the latter should not mutate spans in OpenTelemetry convention.
